src/Core/src/Platform/Android/MauiApplication.cs

```diff
@@ -30,18 +31,20 @@ public override void OnCreate()
 
 			Services = host.Services;
 			Application = Services.GetRequiredService<IApplication>();
-
 			base.OnCreate();
 		}
 
 		// Configure native services like HandlersContext, ImageSourceHandlers etc.. 
 		void ConfigureNativeServices(HostBuilderContext ctx, IServiceCollection services)
 		{
+			services.AddSingleton(services => _windows);
```

Comment:
Since this is on IApplication, is there an issue jetting that and then doing app.Windows?

Reply:
if you have a better idea, by registering here you can get it on your IApplication instance and set ... to return as IReadOnlyCollection. 

on forms Application, i send a message with the window to inject to local collection on maui.controls.Application